.github/workflows/python-avatar.yml

```diff
@@ -0,0 +1,46 @@
+# Build and test the python package
+name: Python Avatar
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Avatar [${{ matrix.shard }}]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        shard: [
+           1/24,  2/24,  3/24,  4/24,
+           5/24,  6/24,  7/24,  8/24,
+           9/24, 10/24, 11/24, 12/24,
+          13/24, 14/24, 15/24, 16/24,
+          17/24, 18/24, 19/24, 20/24,
+          21/24, 22/24, 23/24, 24/24,
+        ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set Up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install
+        run: |
+          pip install --upgrade pip
```

Comment:
I think the recommended invocation for pip in this type of action is `python -m pip` rather than just `pip` (I forget the exact reason, but I remember making that change for the other GitHub actions in this project)

Reply:
Because pip works in global, so when the project runs in a venv, it will fail. Though workflows may run in a container, `python -m venv` still avoid potential failures.

Reply:
Done!
I wasn't aware of that, I'm gonna have a look :+1: